src/cleanfigure.m

```diff
@@ -347,104 +349,162 @@ function simplifyLine(meta, handle, targetResolution)
     yData = get(handle, 'YData');
     zData = get(handle, 'ZData');
 
-    if ~isempty(zData)
-        % TODO: 3d simplificattion of frontal 2d projection
-        return;
-    end
-    if isempty(xData) || isempty(yData)
+    if numel(xData) <= 2 || numel(xData) <= 2
         return;
     end
-    if numel(xData) <= 2
-        return;
+
+    % Check whether this is a 3D plot
+    % Check if elevation is not orthogonal to xy plane
+    [az, el] = view(meta.gca);
+    if el == 90
+        is3D = false;
+    else
+        is3D = true;
     end
 
     % Get info about log scaling
     isXlog = strcmp(get(meta.gca, 'XScale'), 'log');
-    vxData = xData;
     if isXlog
-        vxData = log10(xData);
+        xData = log10(xData);
     end
 
     isYlog = strcmp(get(meta.gca, 'YScale'), 'log');
-    vyData = yData;
     if isYlog
-        vyData = log10(yData);
+        yData = log10(yData);
     end
```

Comment:
We need to differentiate between visual data and the raw unscaled one:  

https://github.com/matlab2tikz/matlab2tikz/commit/bf03aafd4cb6e5b75e909256732014ae8253fd15#commitcomment-10332439